### PR TITLE
Add IPv6 address class

### PR DESCRIPTION
**Pull request by keniachang**
Implement the IPv6 address class aiming to be used to implement the IPv6 source and IPv6 destination match fields as proposed in in of_core: https://github.com/kytos/of_core/issues/77

### CODE REVIEW

File: pyof/foundation/basic_types.py [hunk 1]
```diff
@@ -343,6 +352,131 @@ def __deepcopy__(self, memo):
...
+                start += 1
+
+            final_address = address_temp[:start]
+            for i in range(num_pad):
+                if i + 1 == num_pad:
+                    final_address += '0'
+                else:
+                    final_address += '0:'
+            final_address += address_temp[start:]
+            address = final_address
```
Comment (ajoaoff) [inline pyof/foundation/basic_types.py:397]:
```suggestion
        if address == '::':
            address = '0:0:0:0:0:0:0:0'
        elif '::' in address:
            temp = address.split(':')
            index = temp.index('')
            temp = [x for x in temp if x != '']
            address = temp[:index] + ['0'] * (8 - len(temp)) + temp[index:]
```

File: pyof/foundation/basic_types.py [hunk 2]
```diff
@@ -115,6 +115,15 @@ class UBInt64(GenericType):
     _fmt = "!Q"
 
 
+class UBInt128(GenericType):
+    """Format character for an Unsigned Long Long.
+
+    Class for an 64-bit (8-byte) Unsigned Integer.
```
Comment (gleybersonandrade) [inline pyof/foundation/basic_types.py:121]:
```suggestion
    Class for an 128-bit (16-byte) Unsigned Integer.
```

**Comment by hdiogenes** [comment]
@keniachang, could you provide unit tests for this code? After that, it would be interesting to explore the option of using the built-in [ipaddress](https://docs.python.org/3/library/ipaddress.html#ip-network-definitions) Python library instead of your custom IP address/network parsing.

**Comment by keniachang** [comment]
> @keniachang, could you provide unit tests for this code? After that, it would be interesting to explore the option of using the built-in [ipaddress](https://docs.python.org/3/library/ipaddress.html#ip-network-definitions) Python library instead of your custom IP address/network parsing.

Hi @hdiogenes, sorry I do not have a formal unit test for this code. I just focused on testing the part of expanding the abbreviated IPv6 address by passing four different abbreviated IPv6 addresses one by one and printing the expanded address at the end. The four IPv6 addresses are '::', 'ffff:ca03::c20a', '::ffff:874b:2b34', and 'ffff:ca03:c20a::'.

File: pyof/foundation/basic_types.py [hunk 3]
```diff
@@ -343,6 +352,114 @@ def __deepcopy__(self, memo):
         return IPAddress(address=self._value, netmask=self.netmask)
 
 
+class IPV6Address(GenericType):
+    """Defines a IPV6 address."""
```
Comment (hdiogenes) [inline pyof/foundation/basic_types.py:356]:
```suggestion
class IPv6Address(GenericType):
    """Defines a IPv6 address."""
```

File: pyof/foundation/basic_types.py [hunk 4]
```diff
@@ -343,6 +352,114 @@ def __deepcopy__(self, memo):
...
+
+    netmask = UBInt128()
+
+    def __init__(self, address="0000:0000:0000:0000:0000:0000:0000:0000/128",
+                 netmask=None):
+        """Create an IPV6Address with the parameters below.
```
Comment (hdiogenes) [inline pyof/foundation/basic_types.py:362]:
```suggestion
        """Create an IPv6Address with the parameters below.
```

File: pyof/foundation/basic_types.py [hunk 5]
```diff
@@ -343,6 +352,114 @@ def __deepcopy__(self, memo):
...
+
+        Args:
+            address (str): IP Address using ipv6.
```
Comment (hdiogenes) [inline pyof/foundation/basic_types.py:365]:
```suggestion
            address (str): IP Address using IPv6.
```

**Pull request merged**